### PR TITLE
Me 31

### PR DESCRIPTION
**Pull request by dhiraj-toptal**

### CODE REVIEW

File: Model/Request/Transactions.php [hunk 1]
```diff
@@ -118,19 +118,25 @@ protected function getResponseHandlerType()
...
+                foreach ($termsOptions as $terms) {
+                    $options[$terms] = $terms;
+                }
+            } else {
+                $globalTermsOptions = $this->_balancepayConfig->getTermsOption();
+                if (!empty($globalTermsOptions)) {
+                    foreach ($globalTermsOptions as $terms) {
+                        $options[$terms] = $terms;
+                    }
+                }
```
Comment (ynaxonBalance) [inline Model/Request/Transactions.php]:
```
        $customerId = $this->customerSession->getCustomer()->getId();
        $termsOptions = $this->_balancepayConfig->getMerchantTermsOptions();
        $options = [];
        if ($customerId) {
          $termsOptions = $this->getCustomerTermOptions($customerId) || $termsOptions;
        }
        foreach ($termsOptions as $terms) {
            $options[$terms] = $terms;
        } 
```

File: Model/Config.php [hunk 2]
```diff
@@ -401,6 +401,19 @@ public function getAllowedCustomerGroups($scope = ScopeInterface::SCOPE_STORE, $
...
+    /**
+     * GetMerchantTermsOptions
+     *
+     * @param string $scope
+     * @param int $storeId
+     * @return array|string[]
+     */
+    public function getMerchantTermsOptions($scope = ScopeInterface::SCOPE_STORE, $storeId = null)
+    {
+        return (($apm = $this->getConfigValue('terms_option', $scope, $storeId)) && is_string($apm))
```
Comment (ynaxonBalance) [inline Model/Config.php]: what is `$apm`?

File: Model/Config/Customer/TermsOptionsLimit.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    /**
+     * @return TermsOptionsLimit|void
+     * @throws LocalizedException
+     */
+    public function beforeSave()
+    {
+        $limits = $this->getData('groups/balancepay/fields/terms_option/value') ?? '';
+        if (count($limits) > self::TERMS_LIMIT) {
+            throw new LocalizedException(__('Select upto only 3 terms options.'));
```
Comment (ynaxonBalance) [inline Model/Config/Customer/TermsOptionsLimit.php]: "Select up to ..."

File: Model/Request/Transactions.php [hunk 4]
```diff
@@ -178,14 +180,14 @@ protected function getBuyerParams(Quote $quote)
...
-     * GetTermOptions
+     * GetCustomerTermOptions
      *
      * @param int $customerId
      * @return array|string[]
      * @throws \Magento\Framework\Exception\LocalizedException
      * @throws \Magento\Framework\Exception\NoSuchEntityException
      */
-    public function getTermOptions($customerId)
+    public function getCustomerTermOptions($customerId)
```
Comment (ynaxonBalance) [inline Model/Request/Transactions.php]: `getCustomerTermsOptions` (terms not term)

File: Model/Config.php (hunk 2 above)
Comment (dhiraj-toptal) [reply Model/Config.php]: Its a temporary variable, should I update it to $merchantTermsOptions?

**Comment by dhiraj-toptal** [comment]
Hi @ynaxonBalance  I have updated the PR, can you review?

**Review by ynaxonBalance — APPROVED** [review]

**Pull request merged**